test/unit/host_test.rb

```diff
@@ -3239,6 +3239,28 @@ def to_managed!
     end
   end
 
+  test "should find smart proxy ids" do
+    host_1 = FactoryGirl.create(:host, :with_tftp_orchestration)
+    host_2 = FactoryGirl.create(:host, :with_dns_orchestration)
+    host_3 = FactoryGirl.create(:host, :with_dhcp_orchestration)
+    host_4 = FactoryGirl.create(:host, :with_realm)
+    host_5 = FactoryGirl.create(:host, :with_puppet)
+    host_6 = FactoryGirl.create(:host, :with_puppet_ca)
+
+    tftp_proxy_id = host_1.primary_interface.subnet.tftp_id
+    dns_proxy_id = host_2.primary_interface.subnet.dns_id
+    dhcp_proxy_id = host_3.primary_interface.subnet.dhcp_id
+    realm_proxy_id = host_4.realm.realm_proxy_id
+    puppet_id = host_5.puppet_proxy_id
+    puppet_ca_id = host_6.puppet_ca_proxy_id
+
+    res = Host.smart_proxy_ids(Host.where(:id => [host_1, host_2, host_3, host_4].map(&:id)))
```

Comment:
Please add host_5 and host_6 to the list, I think it's the reason the test doesn't pass.